src/modal.tsx

```diff
@@ -256,10 +256,10 @@ export class ReactNativeModal extends React.Component<ModalProps, State> {
   }
 
   componentWillUnmount() {
-    BackHandler.removeEventListener(
+    BackHandler.addEventListener(
       'hardwareBackPress',
       this.onBackButtonPress,
-    );
+    )?.remove();
```

Comment:
I'm not familiar with this API, but how would this work? Aren't you adding a new event listener and immediately removing it?
Shouldn't we instead store a reference of the listener that attached `onBackButtonPress` when it is attached to the app and then remove it here?

Reply:
Sir, this event is fired when the component unmounts, yes i'll add a reference for this.